py/mpconfig.h

```diff
@@ -500,6 +500,12 @@
 #define MICROPY_COMP_RETURN_IF_EXPR (MICROPY_CONFIG_ROM_LEVEL_AT_LEAST_EXTRA_FEATURES)
 #endif
 
+// Whether to enable extra arguments validation when running native code
+// Requires MICROPY_DYNAMIC_COMPILER and costs 160 bytes (Thumb2)
+#ifndef MICROPY_COMP_EXTRA_VALIDATION
```

Comment:
'COMP' is 'COMPILER' here, right? How does this require MICROPY_DYNAMIC_COMPILER? Still this doesn't seem to describe well what is actually implemented here: adding checks in runtime.c means they get are used whenever running any type of code, be it Python or native or calling the functions directly from C. To me this seems as if it should be just called 'extra argument validation'. But then I'm wondering if this shouldn't be an 'all or nothing' approach. I didn't check in detail, but it seems not unlikely that not all possible cases have been covered yet?

Reply:
The rationale is that the issues these checks handle seem to come only from emitted native code, hence the "dependency" on MICROPY_DYNAMIC_COMPILER and why it is disabled by default.  That should be probably be reworded, as you point out.  If you're not using any native emitters or you already have other error checking in place at the source level then you don't need this.  The name has COMP in it because I didn't find any better stanza for the define in the configuration file and I just followed the naming convention.

It is true this may not cover all possible cases, but it handles all the ones I've bumped into plus a few more found by fuzzing after the first @dpgeorge comment.  I'm quite new to that bit of MicroPython internals after all.